src/tensor_ops/impl_max_axes.rs

```diff
@@ -2,21 +2,29 @@ use super::utils::move_tape_and_add_backward_op;
 use crate::devices::{DeviceReduce, EqAccum, MaxAccum, MulAccum};
 use crate::prelude::*;
 
-/// Reduces dimension `I` of the tensor by gathering the maximum value from that dimension.
+/// Reduces `Axes` of the tensor by gathering the maximum value from that dimension.
 ///
-/// **Pytorch equivalent**: `t.amax(I)`
+/// **Pytorch equivalent**: `t.amax(Axes)`
 ///
 /// **NOTE** This evenly distributes gradients between all equal maximum values, instead
 /// of only exactly 1 value.
 ///
-/// Examples:
+/// Example:
 /// ```rust
 /// # use dfdx::prelude::*;
 /// let t = tensor([[1.0, 2.0, 3.0], [-1.0, -2.0, -3.0]]);
 /// let r: Tensor1D<2> = t.max_axis::<-1>();
 /// assert_eq!(r.data(), &[3.0, -1.0]);
 /// ```
-pub fn max_axis<T: Reduce<Axis<I>>, const I: isize>(mut t: T) -> T::Reduced {
+///
+/// Reducing 2 axes:
+/// ```rust
+/// # use dfdx::prelude::*;
+/// # let t = tensor([[1.0, 2.0, 3.0], [-1.0, -2.0, -3.0]]);
+/// let r: Tensor0D = t.max_axes::<Axes2<0, 1>>();
+/// assert_eq!(r.data(), &3.0);
+/// ```
+pub fn max_axes<T: Reduce<Axes>, Axes>(mut t: T) -> T::Reduced {
```

Comment:
Add `AllAxes`, make `AllAxes` default in these functions, and combine `X` and `X_axes`